### PR TITLE
[BottomNavigation] Expose UIView for a given UITabBarItem.

### PR DESCRIPTION
**Pull request by andrewplai**
This will allow MDCBottomNavigationBar to work with popover / tooltip libraries which require an anchor view for positioning (i.e, if we want to display a tooltip over one of the tabs).

Closes [#4824](https://github.com/material-components/material-components-ios/issues/4824)

### CODE REVIEW

File: components/BottomNavigation/tests/unit/BottomNavigationTests.m [hunk 1]
```diff
@@ -163,4 +163,19 @@ -(void)testTitleVisibility {
   XCTAssert(self.bottomNavBar.itemViews.lastObject.label.isHidden);
 }
 
+-(void)testViewForItem {
```
Comment (romoore) [inline components/BottomNavigation/tests/unit/BottomNavigationTests.m]: Please split this into a test found items (at the correct index) and another for an item that is not found.

File: components/BottomNavigation/tests/unit/BottomNavigationTests.m [hunk 2]
```diff
@@ -163,4 +163,19 @@ -(void)testTitleVisibility {
...
+  UITabBarItem *item1 = [[UITabBarItem alloc] initWithTitle:@"1" image:nil tag:0];
```
Comment (romoore) [inline components/BottomNavigation/tests/unit/BottomNavigationTests.m]: Please mark each section of your test with comments for the "Given" (set-up), "When" (condition being tested), and "Then" (assertions) portions of your test.  See `testSelectedItemAfterReset` for an example in this file.

File: components/BottomNavigation/tests/unit/BottomNavigationTests.m [hunk 3]
```diff
@@ -163,4 +163,19 @@ -(void)testTitleVisibility {
...
+  UITabBarItem *item2 = [[UITabBarItem alloc] initWithTitle:@"2" image:nil tag:0];
+  UITabBarItem *item3 = [[UITabBarItem alloc] initWithTitle:@"3" image:nil tag:0];
+  self.bottomNavBar.items = @[item1, item2, item3];
+
+  XCTAssertEqual([self.bottomNavBar viewForItem:item1], self.bottomNavBar.itemViews[0]);
```
Comment (romoore) [inline components/BottomNavigation/tests/unit/BottomNavigationTests.m]: Let's change these asserts to not require the ordering of `itemViews`. Instead, let's just assert that the returned object is an element of the array and that for two items they are not the same view.  Something like:

```objc
UIView *viewForItem1 = [self.bottomNavBar viewForItem:item1];
UIView *viewForItem2 = [self.bottomNavBar viewForItem:item2];

// Then
XCTAssertNotEqual(viewForItem1, viewForItem2);
XCTAssertTrue([self.bottomNavBar.itemViews containsObject:viewForItem1], @"BottomNavBar.itemViews did not contain the view (%@) returned for UITabBarItem (%@)", viewForItem1, item1);
XCTAssertTrue([self.bottomNavBar.itemViews containsObject:viewForItem2], @"BottomNavBar.itemViews did not contain the view (%@) returned for UITabBarItem (%@)", viewForItem2, item2);

```

**Review by romoore — APPROVED** [review]
Looks good overall. Some feedback on the tests. Thanks for the PR!

File: components/BottomNavigation/tests/unit/BottomNavigationTests.m (hunk 1 above)
Comment (andrewplai) [reply components/BottomNavigation/tests/unit/BottomNavigationTests.m]: Sounds good.

File: components/BottomNavigation/tests/unit/BottomNavigationTests.m (hunk 2 above)
Comment (andrewplai) [reply components/BottomNavigation/tests/unit/BottomNavigationTests.m]: Sounds good.

File: components/BottomNavigation/tests/unit/BottomNavigationTests.m (hunk 3 above)
Comment (andrewplai) [reply components/BottomNavigation/tests/unit/BottomNavigationTests.m]: Sounds good.

**Comment by andrewplai** [comment]
Thanks for the feedback, and I've gone ahead and made the updates to the tests [here](https://github.com/material-components/material-components-ios/pull/5061/commits/5e1bf7a2815036a195382be8beb38bed2c79f74c).

**Review by romoore — APPROVED** [review]
LGTM!

**Comment by romoore** [comment]
I've marked this as merge-when-approved since I'll be offline in a few minutes. Anyone else with write access will be able to merge it once it passes CI builds.

File: components/BottomNavigation/tests/unit/BottomNavigationTests.m [hunk 4]
```diff
@@ -163,4 +163,35 @@ -(void)testTitleVisibility {
   XCTAssert(self.bottomNavBar.itemViews.lastObject.label.isHidden);
 }
 
+-(void)testViewForItemFound {
```
[the remark this answers is not in the document]
Comment (romoore) [reply components/BottomNavigation/tests/unit/BottomNavigationTests.m]: A little late to the party, @material-automation !

**Pull request merged**